myia/anf_ir.py

```diff
@@ -42,8 +66,47 @@ def __init__(self) -> None:
         self.return_: Apply = None
         self.debug = GraphDebug()
 
+    def __str__(self) -> str:
+        return self.debug.force_name()
```

Comment:
`#1` is not very readable, we should probably have something like `Graph(id=1)` for str, and maybe something `<Graph(name=1, parameters=[...], return_=... at ...>` for `__repr__`.

Reply:
`Graph(name=1)` suggests that `graph.name == 1`, but it's `graph.debug.name == 1`. Other than that, I suppose a bit more verbosity would be fine.

Reply:
You can also make it `func 1` or `def 1: x, y -> z`, etc. My point was mostly that the string representation should make clear somehow that this is a function/graph.